lib/kafka_producer.js

```diff
@@ -125,10 +127,13 @@ KafkaProducer.prototype.produce = function produce(topic, message, timeStamp, ca
     }
 
     if (self.restClient) {
-        if (self.batching) {
+        if (skipBatching) {
+            produceMessage = self.getProduceMessage(topic, message, timeStamp, 'binary');
+            self._produce(produceMessage, callback);
+        } else if (self.batching) {
```

Comment:
hygiene: you can refactor this to `if (self.batching && !skipBatch) { } else { }`